src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -531,7 +531,10 @@ private void WriteTo(AnalyzerReference reference, ObjectWriter writer, bool chec
                 // snapshot version for analyzer (since it is based on shadow copy)
                 // we can't send over bits and load analyer from memory (image) due to CLR not being able
                 // to find satellite dlls for analyzers.
-                writer.WriteString(assemblyPath);
+                //
+                // make sure we always normalize assemblyPath to lower case since it comes from Assembly.Location
+                // unlike FullPath which comes from string
+                writer.WriteString(assemblyPath.ToLower());
```

Comment:
I thought about LowerInvariant but here that info is actually used to get assembly loaded and lowered string doesn't need to be persisted across VS session or machine so ToLower seems more right than LowerInvariant